source/beamline/CLS/CLSBeamlineStatus.h

```diff
@@ -42,6 +42,8 @@ class CLSBeamlineStatus : public CLSBiStateGroup
 	/// Returns the beam on value for the given control.
 	double componentBeamOnValue(AMControl *control) const { return controlState1ValueMap_.value(control, -1); }
 
+	/// Returns the beamline status PV
+	AMControl * beamlineStatusPVControl() { return beamlineStatusPVControl_ ? beamlineStatusPVControl_ : this; }
```

Comment:
@dretrex the usage of function `beamlineStatusPVControl()` is in `CLSBeamlneStatusView`:
```
	beamStatusEditor_->setControl(beamlineStatus_->beamlineStatusPVControl());
```
We need to return AMControl to do what we want. The original codes in CLSBeamlineStatusView is
```
	beamStatusEditor_->setControl(beamlineStatus_);
```
which is the case when there is no beam status PV.